mitmproxy/protocol/base.py

```diff
@@ -151,11 +159,16 @@ def disconnect(self):
         """
         self.log("serverdisconnect", "debug", [repr(self.server_conn.address)])
         address = self.server_conn.address
-        source_address = self.server_conn.source_address
         self.server_conn.finish()
         self.server_conn.close()
         self.channel.tell("serverdisconnect", self.server_conn)
-        self.server_conn = models.ServerConnection(address, (source_address.host, 0))
+
+        if self.config.options.spoof_source_address:
```

Comment:
I'm assuming you're referring to the following source_address object:
source_address = self.server_conn.source_address

And in that case, the answer is no. If self.config.options.spoof_source_address is true, 
self.server_conn should be initiated with the client's source address.

In case I've misinterpreted your question please let me know.